### ISSUE TITLE
AtomicReferenceFieldUpdater field in enum cause build time failure

### ISSUE THREAD

**Issue by ziyilin** [oracle/graal#1598]
How to reproduce:

AtomicRFUTest.java
```
public class AtomicRFUTest {
    volatile Integer i;
    
    public AtomicRFUTest(int i) {
        this.i=i;
    }
    
    public static void main(String[] args) {
        UnsafeEnum instance=UnsafeEnum.INSTANCE;
        System.out.println(instance.name());
    }
}
```

UnsafeEnum.java
```
import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;

public enum UnsafeEnum {
    INSTANCE;
    UnsafeEnum(){
        
    }
    
    static AtomicReferenceFieldUpdater<AtomicRFUTest, Integer> OWNER_UPDATER=AtomicReferenceFieldUpdater.newUpdater(AtomicRFUTest.class, Integer.class, "i");
}
```

Compile above code with `native-image -cp bin --initialize-at-build-time=UnsafeEnum AtomicRFUTest`

```
Fatal error: com.oracle.svm.core.util.VMError$HostedError: New atomic field updater found after static analysis
        at com.oracle.svm.core.util.VMError.guarantee(VMError.java:85)
        at com.oracle.svm.core.jdk.AtomicFieldUpdaterFeature.processFieldUpdater(RecomputedFields.java:332)
        at com.oracle.svm.core.jdk.AtomicFieldUpdaterFeature.processObject(RecomputedFields.java:320)
        at com.oracle.graal.pointsto.meta.AnalysisUniverse.replaceObject(AnalysisUniverse.java:532)
        at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.replaceObject(AnalysisConstantReflectionProvider.java:136)
        at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.interceptValue(AnalysisConstantReflectionProvider.java:107)
        at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readValue(AnalysisConstantReflectionProvider.java:100)
        at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readFieldValue(AnalysisConstantReflectionProvider.java:75)
        at com.oracle.svm.hosted.meta.HostedField.readValue(HostedField.java:154)
        at com.oracle.svm.hosted.thread.VMThreadLocalCollector.sortThreadLocals(VMThreadLocalCollector.java:109)
        at com.oracle.svm.hosted.thread.VMThreadMTFeature.beforeCompilation(VMThreadMTFeature.java:255)
        at com.oracle.svm.hosted.NativeImageGenerator.lambda$doRun$2(NativeImageGenerator.java:571)
        at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:63)
        at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:571)
        at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:440)
        at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1386)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)

```

**Comment by ziyilin** [comment]
Don't know which commit fixed this issue, but it's good at 19.3.0.2

**Comment by cstancu** [comment]
Not reproducible anymore.
